cluster-autoscaler/cloudprovider/openshiftmachineapi/machineapi_controller_test.go

```diff
@@ -738,19 +738,27 @@ func TestControllerNodeGroups(t *testing.T) {
 
 func TestControllerNodeGroupsNodeCount(t *testing.T) {
 	type testCase struct {
-		nodeGroups    int
-		nodesPerGroup int
+		nodeGroups            int
+		nodesPerGroup         int
+		expectedNodeGroups    int
+		expectedNodesPerGroup int
 	}
 
 	var testCases = []testCase{{
-		nodeGroups:    0,
-		nodesPerGroup: 0,
+		nodeGroups:            0,
+		nodesPerGroup:         0,
+		expectedNodeGroups:    0,
+		expectedNodesPerGroup: 0,
 	}, {
-		nodeGroups:    1,
-		nodesPerGroup: 0,
+		nodeGroups:            1,
+		nodesPerGroup:         0,
+		expectedNodeGroups:    0,
```

Comment:
+1, verifies a node group with 0 nodes is not considered a node group. The naming is vague in this context since a machineset/machinedeployment will become node group once it qualifies for it. Though as Vikas says and you concluded, the names will change eventually. So, just noting.